### PR TITLE
Simplify Filter Sidebar component and improve scrollbar styles

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

(slightly) Related to #412 by @panchovm

## Description

This PR removes one of the nested Sidebar Filter components to simplify it, and also removes the gap between the sidebar and main content by setting the scrollbar background to transparent. The scrollbar for Filter Sidebar is complicated with the iframe: it now technically doesn't overflow, so the 'thumb' of the scrollbar is invisible, but the gap is still there.
This PR is based on the `VCheckbox` PR.

## Testing Instructions

Run the app and open the Filter sidebar. You should not see this gap between it and the content:
Screen Shot 2021-11-18 at 3 53 59 PM

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/Filters/FilterChecklist.vue [hunk 1]
```diff
@@ -1,147 +1,101 @@
 <template>
-  <div
-    :class="{ ['filters']: true, ['single']: isSingleFilter, 'mb-8': true }"
-    @click="hideLicenseExplanationVisibility()"
-    @keyup.enter="hideLicenseExplanationVisibility()"
-  >
+  <fieldset class="mb-8" @click.stop="hideLicenseExplanationVisibility">
```
Comment (sarayourfriend) [inline src/components/Filters/FilterChecklist.vue:2]: Technically if a non-tabbable element has a click handler you're supposed to add `enter` and `space` as well as a tabindex of 0 to make it accessible, but given this is just a stop-gap solution until we replace it with the VPopover component I think this is okay 🤔

**Comment by zackkrida** [comment]
I don't feel like I can test this, because I'm not able to see the sidebar scrollbar track _or_ a scrollbar for the sidebar _at all_, with the exception of the 'mobile' view:

CleanShot 2021-11-18 at 11 49 55@2x
CleanShot 2021-11-18 at 11 49 51@2x

**Comment by obulat** [comment]
> feel like I can test this, because I'm not able to see the sidebar scrollbar track _or_ a scrollbar for the sidebar _at all_, with the exceptio

Do you mean that _on this branch_ you do not see the scrollbar track? This is a feature, not a bug here :) Since the filter sidebar doesn't scroll (it has the full height here), I thought that I might set all colors to transparent for the transitional styles. Can you see the scrollback track (ie the gap between the filters sidebar and the main content) on the main branch?

File: src/components/Filters/FilterChecklist.vue (hunk 1 above)
Comment (obulat) [reply src/components/Filters/FilterChecklist.vue]: Oh, I didn't realize that it was a non-tabbable element when I removed the `keyup.enter` event handling. And the `tabindex` wasn't there at all. Can't wait for VPopover replacement :)

**Comment by zackkrida** [comment]
> Can you see the scrollback track (ie the gap between the filters sidebar and the main content) on the main branch?

No, I never see it anywhere. Here's the live site for me:

CleanShot 2021-11-19 at 08 18 38@2x

It's because the site doesn't restrict the sidebar height, so there's never a scrollbar!

This work should be very useful however when we can switch back to a fixed 100vh-height sidebar with its own scrolling.

**Comment by sarayourfriend** [comment]
I wonder if there's some kind of configuration difference? I always see it in my Firefox browser.

Captura de Tela 2021-11-19 às 07 25 10

Safari too

Captura de Tela 2021-11-19 às 07 25 54

I even opened Chrome for you 😱 and it shows it for me as well.

Captura de Tela 2021-11-19 às 07 26 27

**Comment by griffbrad** [comment]
Possible you have different “Show scroll bars” settings in MacOS? (https://support.apple.com/guide/mac-help/change-general-preferences-mchlp1225/mac)

**Review by zackkrida — APPROVED** [review]
Lovely work; the removal of the `FiltersList` is great. 

As I mentioned, I now am able to see the scrollbar track when using my mac in clamshell mode with an external mouse and keyboard 🤔

**Review by sarayourfriend — APPROVED** [review]
LGTM!

**Pull request merged**